lib/checkother.cpp

```diff
@@ -1550,11 +1550,13 @@ void CheckOther::checkConstPointer()
             continue;
         const Token* const nameTok = tok->variable()->nameToken();
         // declarations of (static) pointers are (not) split up, array declarations are never split up
-        if (tok == nameTok && (!Token::simpleMatch(tok->variable()->typeStartToken()->previous(), "static") || Token::simpleMatch(nameTok->next(), "[")))
+        if (tok == nameTok && (!Token::simpleMatch(tok->variable()->typeStartToken()->previous(), "static") || Token::simpleMatch(nameTok->next(), "[")) &&
```

Comment:
sorry it's not your code .. but would it work to replace `Token::simpleMatch(tok->variable()->typeStartToken()->previous(), "static")` with `tok->variable()->isStatic()` ?

Reply:
Actually, that is my code... Changed it accordingly.